### PR TITLE
✨ NEW: Add semantic HTML roles

### PR DESCRIPTION
**Pull request by rowanc1**
Added a few of the basic html roles that are in sphinx out of the box.

- [x] abbr
- [x] sup
- [x] sub
- [x] math

### CODE REVIEW

**Comment by rowanc1** [comment]
I think the translation/resolution of roles into various languages is likely something that can be kicked down the road for a while. The logic for this is combined to two, non-exposed, places (in the Role/Directive classes) currently the logic is to look into a user supplied dictionary, that can always get more sophisticated if the feature is necessary.

I will add the three aliases that you suggest.

I will also add the other math role in this PR, and then get it into main - so we can resolve some of the math rendering in a similar way between the various repositories.

**Comment by rowanc1** [comment]
Just another note, `abbr` was likely chosen because it is the [html standard](https://www.w3schools.com/tags/tag_abbr.asp)!

**Comment by rowanc1** [comment]
Ok - this PR is good to go in from my perspective.

File: src/roles/plugin.ts [hunk 1]
```diff
@@ -7,15 +7,16 @@ import type StateCore from "markdown-it/lib/rules_core/state_core"
 import type StateInline from "markdown-it/lib/rules_inline/state_inline"
 import type Token from "markdown-it/lib/token"
 import { Role } from "./main"
+import { inlineMathRenderer, IOptions as IMathRoleOptions } from "./math"
 
 /** Allowed options for directive plugin */
-export interface IOptions {
+export interface IOptions extends IMathRoleOptions {
```
Comment (chrisjsewell) [inline src/roles/plugin.ts:13]: I don't think this is the correct way to go; having inheritance add role based options (or indeed any options). We need to come up with a better way of doing this

Comment (chrisjsewell) [reply src/roles/plugin.ts:13]: Off hand, I would suggest something like:

```typescript
export interface IOptions {
  suboptions: {
    math: {
      renderer?: (content: string, opts: { displayMode: boolean }) => string
    }
  }
}
```

**Review by chrisjsewell — CHANGES_REQUESTED** [review]
See comment about math render option.

Plus, can you also add these roles to the list in the README, and maybe also to the docs/index.html initial input text

File: src/roles/plugin.ts (hunk 1 above)
Comment (rowanc1) [reply src/roles/plugin.ts]: I have adressed this!

**Comment by rowanc1** [comment]
This should be good to go in.

File: src/directives/plugin.ts [hunk 2]
```diff
@@ -23,11 +12,11 @@ export default function directivePlugin(md: MarkdownIt, options: IOptions): void
   md.core.ruler.after(after, "run_directives", runDirectives(options.directives || {}))
 
   // fallback renderer for unhandled directives
-  md.renderer.rules["directive"] = (tokens: Token[], idx: number) => {
```
Comment (chrisjsewell) [inline src/directives/plugin.ts:26]: minor question: how comes you removed the types here?

File: src/roles/html.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import type Token from "markdown-it/lib/token"
```
Comment (chrisjsewell) [inline src/roles/html.ts:1]:
```suggestion
/**
 * This module contains roles that directly map to HTML semantic tags
 */
import type Token from "markdown-it/lib/token"
```

File: src/roles/math.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import type MarkdownIt from "markdown-it/lib"
```
Comment (chrisjsewell) [inline src/roles/math.ts:1]:
```suggestion
/**
 * This module contains roles that relate to mathematics
 */
import type MarkdownIt from "markdown-it/lib"
```

**Review by chrisjsewell — CHANGES_REQUESTED** [review]
Just a nitpick: I'm very pro documenting modules/classes/functions/attributes etc, we are in the business of documentation after all, lol

**Review by chrisjsewell — APPROVED** [review]

**Pull request merged**